plugins/inputs/t128_peer_path/t128_peer_path.go

```diff
@@ -0,0 +1,183 @@
+	gqlCollector     *t128_graphql.T128GraphQL
+	endpointNotFound bool
+}
+
+var peerPathEntryPoint = "allPeers/nodes/paths"
+
+var peerPathFields = map[string]string{"status": "status", "enabled": "enabled"}
+
+var peerPathTags = map[string]string{
+	"node":             "node",
+	"adjacentAddress":  "adjacentAddress",
+	"deviceInterface":  "deviceInterface",
+	"networkInterface": "networkInterface",
+	"vlan":             "vlan",
+	"routerName":       "allPeers/nodes/routerName",
+}
+
+var peerPathHostnameTags = map[string]string{
+	"node":             "node",
+	"adjacentAddress":  "adjacentAddress",
+	"adjacentHostname": "adjacentHostname",
+	"deviceInterface":  "deviceInterface",
+	"networkInterface": "networkInterface",
+	"vlan":             "vlan",
+	"routerName":       "allPeers/nodes/routerName",
+}
+
+var sampleConfig = `
+[[inputs.t128_peer_path]]
+## Required. A name for the collector which will be used as the measurement name of the produced data.
+# collector_name = "peer-path-state"
+
+## Required. The base url for data collection.
+## graphQL ports vary across 128T versions.
+# base_url = "http://localhost:31517/api/v1/graphql/"
+
+## A socket to use for retrieving data - unused by default
+# unix_socket = "/var/run/128technology/web-server.sock"
+
+## Amount of time allowed before the client cancels the HTTP request
+# timeout = "5s"
+
+## Required. The versioned field to include hostname or not
+# exclude_hostname = false
+`
+
+// SampleConfig returns the default configuration of the Input
+func (*T128PeerPath) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns a one-sentence description on the Input
+func (*T128PeerPath) Description() string {
+	return "Make a 128T Peer Path query and return the data"
+}
+
+// Init sets up the input to be ready for action
+func (plugin *T128PeerPath) Init() error {
+
+	var query string
+	var tags map[string]string
+	if plugin.ExcludeHostname {
+		query = peerPathQuery
+		tags = peerPathTags
+	} else {
+		query = peerPathHostnameQuery
+		tags = peerPathHostnameTags
+	}
+	plugin.gqlCollector = &t128_graphql.T128GraphQL{
+		CollectorName:   plugin.CollectorName,
+		BaseURL:         plugin.BaseURL,
+		UnixSocket:      plugin.UnixSocket,
+		EntryPoint:      peerPathEntryPoint,
+		Fields:          peerPathFields,
+		Tags:            tags,
+		Timeout:         plugin.Timeout,
+		RetryIfNotFound: plugin.RetryIfNotFound,
+		Query:           query,
+	}
+	return plugin.gqlCollector.Init()
+}
+
+// Gather takes in an accumulator and adds the metrics that the Input gathers
+func (plugin *T128PeerPath) Gather(acc telegraf.Accumulator) error {
+	if !plugin.RetryIfNotFound && plugin.endpointNotFound {
+		return nil
+	}
+	processedResponses, err := plugin.gqlCollector.MakeRequest()
+	if err != nil {
+		for _, err := range err {
+			acc.AddError(err)
+		}
+	}
+	for _, processedResponse := range processedResponses {
+		for k, v := range processedResponse.Tags {
```

Comment:
okay - this answers my question about why did we chose to run this as a native collector as opposed to using the graphql input. I guess its fine since this logic is highly custom.